conversational-client/src/app/app.component.ts

```diff
@@ -51,14 +64,23 @@ export class AppComponent implements OnInit {
     });
   }
 
+  ngAfterViewChecked() {
+    // Initialize the agent registries once child views are ready.
+    this.agentRegistry = {
```

Comment:
(apologies if this is WIP) -- feels like this should be pulled out into a separate file so that anyone building an agent doesn't need to change app.component.

Also not sure why DefaultAgent is different to the others?

Reply:
Done - I moved the registry to agents.service